validator/src/journal/journal_ffi.rs

```diff
@@ -67,11 +72,28 @@ use py_object_wrapper::PyObjectWrapper;
 
 struct Journal {
     pub chain_controller: ChainController,
+    pub genesis_controller: GenesisController,
 }
 
 impl Journal {
-    fn start(&mut self) {
+    fn start(&mut self) -> ErrorCode {
+        if match self.genesis_controller.requires_genesis() {
+            Ok(bool) => bool,
+            Err(err) => {
+                error!(
+                    "Unable to check if the genesis block should be built: {}",
+                    err
+                );
+                return ErrorCode::Unknown;
+            }
+        } {
+            if let Err(err) = self.genesis_controller.start() {
```

Comment:
Would be a bit simpler if you moved this to an `Ok(true)` arm of the match above

Reply:
updated